electron-app/src/server/websites/mastodon/mastodon.service.ts

```diff
@@ -276,64 +288,59 @@ export class Mastodon extends Website {
+    const M = generator('mastodon', accountData.website, accountData.token);
     const instanceInfo: MastodonInstanceInfo = this.getAccountInfo(data.part.accountId, INFO_KEY);
     const maxChars = instanceInfo ? instanceInfo?.configuration?.statuses?.max_characters : 500;
 
     const isSensitive = data.rating !== SubmissionRating.GENERAL;
 
     const { options } = data;
-    const form: any = {
-      status: `${options.useTitle && data.title ? `${data.title}\n` : ''}${data.description}`,
+    let status = `${options.useTitle && data.title ? `${data.title}\n` : ''}${data.description}`;
+    const statusOptions: any = {
       sensitive: isSensitive,
       visibility: options.visibility || 'public',
+      spoiler_text: ""
     };
 
-    const tags = this.formatTags(data.tags);
+    const tags: string[] = this.parseTags(data.tags, {
+      spaceReplacer: '_',
+      minLength: 1,
+      maxLength: 100,
+    });
 
-    // Update the post content with the Tags if any are specified - for Mastodon, we need to append
+    // Update the post content with the Tags if any are specified - for Mastodon, we need to append 
     // these onto the post, *IF* there is character count available.
     if (tags.length > 0) {
-      form.status += '\n\n';
+      status += "\n\n";
     }
 
-    tags.forEach((tag) => {
-      let remain = maxChars - form.status.length;
+    tags.forEach(tag => {
+      let remain = maxChars - status.length;
       let tagToInsert = tag;
-      if (remain > tagToInsert.length) {
-        form.status += ` ${tagToInsert}`;
+      if (!tag.startsWith('#')) {
+        tagToInsert = `#${tagToInsert}`
+      }
+      if (remain > (tagToInsert.length)) {
+        status += ` ${tagToInsert}`
       }
       // We don't exit the loop, so we can cram in every possible tag, even if there are short ones!
-    });
+    })
 
     if (options.spoilerText) {
-      form.spoiler_text = options.spoilerText;
+      statusOptions.spoiler_text = options.spoilerText;
     }
 
     this.checkCancelled(cancellationToken);
-    const post = await M.post('statuses', form);
-
-    if (!post.data.id || post.data.error) {
+    
+    M.postStatus(status, statusOptions).then((result) => {
+      let res = result.data as Entity.Status;
+      return this.createPostResponse({ source: res.url });
+    }).catch((err: Error) => {
       return Promise.reject(
-        this.createPostResponse({ message: post.data.error, additionalInfo: post.data }),
+        this.createPostResponse({ message: err.message }),
       );
-    }
-
-    return this.createPostResponse({ source: post.data.url });
-  }
-
-  formatTags(tags: string[]) {
-    return this.parseTags(
-      tags
-        .map((tag) => tag.replace(/[^a-z0-9]/gi, ' '))
-        .map((tag) =>
-          tag
-            .split(' ')
-            // .map((word) => word.charAt(0).toUpperCase() + word.slice(1))
-            .join(''),
-        ),
-      { spaceReplacer: '' },
-    ).map((tag) => `#${tag}`);
+    })
```

Comment:
Needs a return or await.
This would just immediately fall through to line 343.

Reply:
Resolved